pkg/auth/providers/keycloakoidc/keycloak_client.go

```diff
@@ -64,31 +64,54 @@ func (k *KeyCloakClient) searchPrincipals(searchTerm, principalType string, conf
 		}
 	}
 	if principalType == "" || principalType == GroupType {
-		var groups []Group
-		searchURL := fmt.Sprintf("%s/%ss?search=%s", sURL, GroupType, searchTerm)
-		search := URLEncoded(searchURL)
-
-		b, statusCode, err := k.getFromKeyCloak(search)
+		groupAccounts, err := k.groupSearch(searchTerm, sURL)
 		if err != nil {
-			logrus.Errorf("[keycloak oidc]: GET request failed, got status code: %d. url: %s, err: %s",
-				statusCode, search, err)
-			return accounts, err
-		}
-		if err := json.Unmarshal(b, &groups); err != nil {
-			logrus.Errorf("[keycloak oidc]: received error unmarshalling search results, err: %v", err)
 			return accounts, err
 		}
-		for _, g := range groups {
-			accounts = append(accounts, account{ID: g.ID, Name: g.Name, Type: GroupType})
-			subGroups := getSubGroups(g)
-			for _, sg := range subGroups {
-				accounts = append(accounts, account{ID: sg.ID, Name: sg.Name, Type: GroupType})
-			}
+		accounts = append(accounts, groupAccounts...)
+	}
+	if err != nil {
+		return accounts, nil
+	}
+	return accounts, nil
+}
+
+func (k *KeyCloakClient) groupSearch(searchTerm string, sURL string) ([]account, error) {
+	var groups []Group
+	var accounts []account
+
+	searchURL := fmt.Sprintf("%s/%ss?search=%s", sURL, GroupType, searchTerm)
+	search := URLEncoded(searchURL)
+
+	b, statusCode, err := k.getFromKeyCloak(search)
+	if err != nil {
+		logrus.Errorf("[keycloak oidc]: GET request failed, got status code: %d. url: %s, err: %s",
+			statusCode, search, err)
+		return accounts, err
+	}
+	if err := json.Unmarshal(b, &groups); err != nil {
+		logrus.Errorf("[keycloak oidc]: received error unmarshalling search results, err: %v", err)
+		return accounts, err
+	}
+	for _, g := range groups {
+		accounts = append(accounts, account{ID: g.ID, Name: g.Name, Type: GroupType})
```

Comment:
are groups limited to a depth of 2?

Reply:
https://github.com/rancher/rancher/blob/122aa56e845eec8c0be37717556dd7e1ebebe16a/pkg/auth/providers/keycloakoidc/keycloak_client.go#L98 I added this limit so at most there would be a limit of 100 subgroups per group/subgroup